plugins/woocommerce/src/Blocks/BlockTypes/ProductSaleBadge.php

```diff
@@ -110,8 +110,10 @@ protected function render( $attributes, $content, $block ) {
 			return null;
 		}
 
+		$classname = isset( $attributes['className'] ) ? $attributes['className'] : '';
+		unset( $attributes['className'] );
+
```

Comment:
I think that we can improve this code, making it this logic more explicit (other developers could not know how `StyleAttributesUtils` works).

`StyleAttributesUtils` accepts as third parameter the exclude option.


```suggestion
// Add global style classes and an extra class name to different DOM elements.
		$classes_and_styles = StyleAttributesUtils::get_classes_and_styles_by_attributes( $attributes, array(), 'extra_classes' );
		$classname = StyleAttributesUtils::get_classes_and_styles_by_attributes($attributes, array('extra_classes'))

```

Reply:
Agree, I totally missed the third parameter.